File: woodroid/src/main/java/me/gilo/woodroid/models/filters/CustomerFilter.java
package me.gilo.woodroid.models.filters;

public class CustomerFilter extends ListFilter{

    String email;
    String role;

    //all, administrator, editor, author, contributor, subscriber, customer and shop_manager

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
        addFilter("email", email);
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
        addFilter("role", role);
    }

    public void setRole(Role role) {
        this.role = role.toString();
        addFilter("role", role.toString());
    }
}

File: woodroid/src/main/java/me/gilo/woodroid/models/filters/CustomerFilter.kt
package me.gilo.woodroid.models.filters

class CustomerFilter : ListFilter() {

    private lateinit var email: String
    private lateinit var role: String

    //all, administrator, editor, author, contributor, subscriber, customer and shop_manager

    fun getEmail(): String {
        return email
    }

    fun setEmail(email: String) {
        this.email = email
        addFilter("email", email)
    }

    fun getRole(): String {
        return role
    }

    fun setRole(role: String) {
        this.role = role
        addFilter("role", role)
    }

    fun setRole(role: Role) {
        this.role = role.toString()
        addFilter("role", role.toString())
    }
}
